packages/jest-cli/src/cli/processArgs.js

```diff
@@ -221,8 +223,26 @@ function processArgs() {
       if (argv.testEnvData) {
         argv.testEnvData = JSON.parse(argv.testEnvData);
       }
+
+      const yargsSpecialOptions = ['$0', '_'];
+      const allowedOptions = Object.keys(options).reduce((acc, option) => {
+        return acc
+          .add(option)
+          .add(options[option].alias);
+      }, new Set(yargsSpecialOptions));
+      const unrecognizedOptions = Object.keys(argv).filter(arg => {
+        return !allowedOptions.has(arg);
+      });
+      if (unrecognizedOptions.length) {
+        throw new Error(
```

Comment:
right, but this is gonna go into a patch release where I don't want to break everyone. If it runs 1000 test, it'll tell you and you can always kill it with ctrl+c.

It also makes it easier switching between two versions when adding a new arg, something that I do a lot and why I changed the configs to warn instead of throw as well. There isn't really much point in not running any tests if something passed to the runner cannot be understood.